packages/components/src/date-time/test/time.js

```diff
@@ -271,4 +271,28 @@ describe( 'TimePicker', () => {
 
 		expect( monthInputIndex > dayInputIndex ).toBe( true );
 	} );
+
+	it( 'Should set a time when passed a null currentTime', () => {
+		const onChangeSpy = jest.fn();
+
+		render(
+			<TimePicker
+				currentTime={ null }
+				onChange={ onChangeSpy }
+				is12Hour
+			/>
+		);
+
+		const monthInput = screen.getByLabelText( 'Month' ).value;
+		const dayInput = screen.getByLabelText( 'Day' ).value;
+		const yearInput = screen.getByLabelText( 'Year' ).value;
+		const hoursInput = screen.getByLabelText( 'Hours' ).value;
+		const minutesInput = screen.getByLabelText( 'Minutes' ).value;
+
+		expect( isNaN( monthInput ) ).toBe( false );
+		expect( isNaN( dayInput ) ).toBe( false );
+		expect( isNaN( yearInput ) ).toBe( false );
+		expect( isNaN( hoursInput ) ).toBe( false );
+		expect( isNaN( minutesInput ) ).toBe( false );
```

Comment:
Nitpicking: Prefer using [`Number.isNaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN) here as it's a more robust version. It shouldn't matter in this case though 😅.

Reply:
I deliberately used `isNaN` because `Number.isNaN` will return false for any string. We need to check that the value is either a number, or a string that can be parsed into a number, as opposed to something like 'invalid date'. The problem with this approach is that it won't catch an empty string or a `null` value, but that won't change even if we use `Number.isNaN`. It might be worth calling `parseInt` on those variables to make sure anything that's not a number is actually flagged as a NaN 😅

Reply:
... and in that case we _can_ use `Number.isNaN` 😸